lib/charms/prometheus_k8s/v0/prometheus_scrape.py

```diff
@@ -335,6 +340,81 @@ def _on_scrape_targets_changed(self, event):
 }
 
 
+RELATION_INTERFACE_NAME = "prometheus_scrape"
+
+
+class RelationDirection(Enum):
+    """Relations can be either provided or required by a charm."""
+
+    PROVIDED = 1
+    REQUIRED = 2
```

Comment:
Reminds me of Java :)
I would replace `relation_direction: RelationDirection` with `relation_direction: typing.Literal["provides", "requires"]` so the enum wouldn't be needed.

Reply:
Turns out there is already `ops.charm.RelationRole` ;-)